src/Model/ViewModel.php

```diff
@@ -76,7 +76,7 @@ class ViewModel implements ModelInterface, ClearableModelInterface, RetrievableC
     /**
      * Constructor
      *
-     * @param  null|array<string, mixed>|Traversable<string, mixed>|ArrayAccess<string, mixed> $variables
+     * @param  null|array<string, mixed>|Traversable<string, mixed>|ArrayAccess<string, mixed>|array<int, mixed>|Traversable<int, mixed>|ArrayAccess<int, mixed> $variables
```

Comment:
I'd say that all the `int`-keyed values should be rejected.

Yes, it works in practice: no, it shouldn't be allowed, and there's no guarantee that it will.

Reply:
@Ocramius ok, do you have a suggestion of how to implement this?
I see a few possibilities here, but ultimately it leads me to thinking that `JsonModel` shouldn't inherit from `ViewModel` and that `ModelInterface` contains a lot of stuff that is not necessary for `JsonModel`...

Reply:
@ppaulis `JsonModel` could declare its own `__construct`, with more types (`mixed` comes to mind, to be honest)

Reply:
@Ocramius so we continue relying on the same code with new tests for JsonModel specifically (haven't deep-dived into the existing tests yet), testing the discussed JSON format. And when the day comes that the ViewModel would no longer support integer-indexes, we would need to think about how to keep the JsonModel running.

if that's ok, I'll send you over a new PR until end of the week.

Reply:
@gsteel 
What are you planning for version 3.0 on this?

Reply:
Rendering Strategies are, IMO, MVC specific - it's been a while since I had time to allocate to view, but I think my initial plan was to move the rendering strategies to the as yet un-released `laminas-mvc-view` component.